File: LeetCodeJava/70.爬楼梯.java
/*
 * @Author: Xun Zhao
 * @Date: 2019-09-21 16:43:40
 * @LastEditors: Xun Zhao
 * @LastEditTime: 2019-09-21 16:46:36
 * @Description: 
 */
/*
 * @lc app=leetcode.cn id=70 lang=java
 *
 * [70] 爬楼梯
 */
class Solution {
    public int climbStairs(int n) {
        int[] dp = new int[n + 1];
        dp[0] = 1;
        dp[1] = 1;
        for (int i = 2; i < n + 1; i ++) {
            dp[i] = dp[i - 1] + dp[i - 2];
        }
        return dp[n];
    }
}

File: LeetCodePython/70.爬楼梯.py
#
# @lc app=leetcode.cn id=70 lang=python3
#
# [70] 爬楼梯
#

# @lc code=start
class Solution:
    def climbStairs(self, n: int) -> int:
        dp = [1, 1]
        for i in range(1, n):
            dp.append(dp[-1] + dp[-2])
        return dp[-1]

# @lc code=end

